anyway/infographics_utils.py

```diff
@@ -35,18 +96,40 @@
                  'rank': int (Integer)
                  }
     }
-"""
+    """
+    def __init__(self, request_params: RequestParams):
+        self.request_params: RequestParams = request_params
+        self.name: str = ""
+        self.rank: int = -1
+        self.items: Dict = {}
+        self.text: Dict = {}
+        self.meta: Optional[Dict] = None
+        # noinspection PyTypeChecker
+        self.widget_id: WidgetId = None
```

Comment:
I think you can do `self.widget_id: Optional[WidgetId] = None` to get rid off the noinspection error.

Reply:
I want to avoid cases where `widget_id` is not set by mistake, so I prefer to live with the `noinspection` comment

Reply:
I don't understand. Why initialize to a value that's diallowed by the type? It's better not to initialize at all.

Reply:
Add `widget_id` as a mandatory parameter to this constructor, and pass it from subclasses.

Reply:
Good idea

Reply:
Done. It is much better now.